source/extensions/filters/http/ext_proc/processor_state.cc

```diff
@@ -62,7 +62,7 @@ bool ProcessorState::handleHeadersResponse(const HeadersResponse& response) {
       clearWatermark();
 
     } else {
-      if (body_mode_ == ProcessingMode::BUFFERED) {
+      if (body_mode_ == ProcessingMode::BUFFERED && !no_body_) {
```

Comment:
Maybe a `if (no_body_) { // do nothing }` branch would make this slightly easier to grok?

Reply:
I agree that it's a little clearer -- take a look.